src/V3EmitCFunc.h

```diff
@@ -1339,7 +1339,14 @@ class EmitCFunc VL_NOT_FINAL : public EmitCConstInit {
     }
     void visit(AstCCast* nodep) override {
         // Extending a value of the same word width is just a NOP.
-        if (const AstClassRefDType* const classDtypep = VN_CAST(nodep->dtypep(), ClassRefDType)) {
+        AstNodeDType* dtypep = nodep->dtypep();
+        if (const AstRefDType* const refDtypep = VN_CAST(dtypep, RefDType)) {
+            dtypep = refDtypep->refDTypep();
+        }
+        if (const AstParamTypeDType* const paramDType = VN_CAST(dtypep, ParamTypeDType)) {
+            dtypep = paramDType->dtypep();
+        }
```

Comment:
But a ref could point to a ref.

Please use ->skipRefp(), or one of the related functions, shouldn't need to roll-your-own here.

Thanks for fixing!